assets/app.js

```diff
@@ -9,4 +9,24 @@ const tooltipTriggerList = document.querySelectorAll('[data-bs-toggle="tooltip"]
 window.addEventListener('load', () => {
   [...tooltipTriggerList].map(tooltipTriggerEl => new bootstrap.Tooltip(tooltipTriggerEl))
   window.dispatchEvent(new Event('ajaxload'))
+
+  const recipientSelector = '.digital-post-recipient'
+  const digitalPostRecipientWrapper = document.querySelector(recipientSelector)?.parentNode?.parentNode
+  const digitalPostAttachmentsInfo = document.querySelector('.digital-post-attachments-info')
+  if (digitalPostRecipientWrapper && digitalPostAttachmentsInfo) {
+    const recipients = digitalPostRecipientWrapper.querySelectorAll(recipientSelector)
+    const updateStuff = () => {
+      const recipientTheCannotReceiveDigitalPost = [...recipients]
+        .filter(el => !el.dataset.digitalPostAllowed &&
+                el.querySelector('input[type="checkbox"]').checked);
+
+      [...digitalPostAttachmentsInfo.querySelectorAll('.digital-post-attachments-info-item')]
+        .forEach(el => {
+          el.hidden = !(el.dataset.digitalPostAllowed ^ (recipientTheCannotReceiveDigitalPost.length > 0))
```

Comment:
If i understand correctly this handles which div should be hidden and which should not in `digital_post_attachments_info.html.twig`, but it took me a while to get my head around.